jbmc/unit/java-testing-utils/require_type.cpp

```diff
@@ -54,40 +54,40 @@ struct_union_typet::componentt require_type::require_component(
 
 /// Checks a type is a code_type (i.e. a function)
 /// \param type: The type to check
-/// \return The cast version of the type code_type
-code_typet require_type::require_code(const typet &type)
+/// \return The cast version of the type method_type
+java_method_typet require_type::require_code(const typet &type)
 {
   REQUIRE(type.id() == ID_code);
-  return to_code_type(type);
+  return to_java_method_type(type);
 }
 
-/// Verify a given type is an code_typet, and that the code it represents
+/// Verify a given type is an java_method_typet, and that the code it represents
 /// accepts a given number of parameters
 /// \param type The type to check
-/// \param num_params check the the given code_typet expects this
+/// \param num_params check the the given java_method_typet expects this
 /// number of parameters
-/// \return The type cast to a code_typet
-code_typet
+/// \return The type cast to a java_method_typet
+java_method_typet
 require_type::require_code(const typet &type, const size_t num_params)
 {
-  code_typet code_type = require_code(type);
-  REQUIRE(code_type.parameters().size() == num_params);
-  return code_type;
+  java_method_typet method_type = require_code(type);
+  REQUIRE(method_type.parameters().size() == num_params);
+  return method_type;
 }
 
 /// Verify that a function has a parameter of a specific name.
 /// \param function_type: The type of the function
 /// \param param_name: The name of the parameter
 /// \return: A reference to the parameter structure corresponding to this
 /// parameter name.
-code_typet::parametert require_type::require_parameter(
-  const code_typet &function_type,
+java_method_typet::parametert require_type::require_parameter(
```

Comment:
I probably wouldn't change this at all - it isn't specific to java methods to have parameters.